### PR TITLE
Examples xcoreai

### PR DESCRIPTION
**Pull request by danielpieczko**
Updated all the examples to run on the xcore.ai MCAB:
- loopback: uses the data generation/checking algorithm that was already there, but uses the shared buffer with the adat_tx_port() API
- rx: receives frames of samples and prints a message regularly to report how many frames have been received
- tx/tx_direct: transmits sine waves with different frequencies and phases across the channels

### CODE REVIEW

File: examples/app_adat_loopback/src/main.xc [hunk 1]
```diff
@@ -1,229 +1,148 @@
...
+// Found solution: IN 24.000MHz, OUT 24.576000MHz, VCO 2457.60MHz, RD 1, FD 102.400 (m = 2, n = 5), OD 5, FOD 5, ERR 0.0ppm
+// Measure: 100Hz-40kHz: ~8ps
+// 100Hz-1MHz: 63ps.
+// 100Hz high pass: 127ps.
+#define APP_PLL_CTL_24M  0x0A006500
+#define APP_PLL_DIV_24M  0x80000004
+#define APP_PLL_FRAC_24M 0x80000104
+
+// Set secondary (App) PLL control register
+void set_app_pll_init (tileref tile, int app_pll_ctl)
```
Comment (xross) [inline examples/app_adat_loopback/src/main.xc]: This should come from lib_sw_pll now ideally. It now has the "fixed" clock settings.

Comment (danielpieczko) [reply examples/app_adat_loopback/src/main.xc]: Yes, good point - I just copied this from the examples in lib_spdif. I'll add lib_sw_pll.

File: examples/app_adat_tx_example/src/main.xc [hunk 2]
```diff
@@ -1,55 +1,152 @@
...
-    set_clock_src(mck_blk, mck);
-    set_clock_fall_delay(mck_blk, 7);   // XAI2 board, set to appropriate value for board.
-    set_port_clock(adat_port, mck_blk);
-    start_clock(mck_blk);
+buffered out port:32 p_adat_tx = PORT_ADAT_OUT;
+in port p_mclk_in = PORT_MCLK_IN;
+out port p_ctrl = PORT_CTRL;
+on tile[1]: clock clk_audio = XS1_CLKBLK_2;
+
+// Found solution: IN 24.000MHz, OUT 24.576000MHz, VCO 2457.60MHz, RD 1, FD 102.400 (m = 2, n = 5), OD 5, FOD 5, ERR 0.0ppm
```
Comment (xross) [inline examples/app_adat_tx_example/src/main.xc]: Get all this from lib_sw_pll

File: examples/app_adat_loopback/src/main.xc (hunk 1 above)
Comment (xross) [reply examples/app_adat_loopback/src/main.xc]: Ah, we should add an issue to lib_spdif to get those moved too then!

**Review by xross — APPROVED** [review]
Looks good, please update changelog before merge.

**Pull request merged**